src/main/java/carpet/script/value/EntityValue.java

```diff
@@ -1294,6 +1319,30 @@ else if (v instanceof ListValue)
             if(e instanceof PlayerEntity) ((PlayerEntity) e).getHungerManager().addExhaustion((int) NumericValue.asNumber(v).getLong());
         });
 
+        put("absorption",(e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).setAbsorptionAmount((float) NumericValue.asNumber(v).getLong());
+        });
+
+        put("add_xp",(e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).addExperience((int) NumericValue.asNumber(v).getLong());
+        });
+
+        put("add_xp_level",(e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).addExperienceLevels((int) NumericValue.asNumber(v).getLong());
+        });
+
+        put("score", (e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).setScore((int) NumericValue.asNumber(v).getLong());
+        });
+
+        put("add_score",(e, v)->{
+            if(e instanceof PlayerEntity) ((PlayerEntity) e).addScore((int) NumericValue.asNumber(v).getLong());
+        });
+        /*
+        put("xp_points",(e, v)->{//idk how useful tbh, and dont work anyways.
+            if(e instanceof PlayerEntity((PlayerEntity) e).experienceProgress=(int) NumericValue.asNumber(v).getLong();
+        });
+        */
```

Comment:
This doesn't work (as far as I understood) because `experienceProgress` is only for getting the progress bar to be nice, but it is reset whenever some change to XP happens (it's not used to derivate new XP values or anything).

Reply:
ye I realised.